gh-100926: Move ctype's pointers cache to StgInfo (#131282)
python/cpython: Doc/whatsnew/3.14.rst

  loaded by the current process.
  (Contributed by Brian Ward in :gh:`119349`.)

* Move :func:`ctypes.POINTER` types cache from the global ``_pointer_type_cache``
  to the corresponding :mod:`ctypes` types. This will stop the cache from
  growing without limits in some situations.

>   (Contributed by Sergey Miryanov in :gh:`100926`).

Author: @neonene How I can name you here?

Reviewer: Please go ahead without me there, thanks.

  loaded by the current process.
  (Contributed by Brian Ward in :gh:`119349`.)

> * Move :func:`ctypes.POINTER` types cache from the global ``_pointer_type_cache``
>   to the corresponding :mod:`ctypes` types. This will stop the cache from
>   growing without limits in some situations. Also :mod:`ctypes` pointer types
>   can be accessed with ``__pointer_type__`` attribute from corresponding type.
>   (Contributed by Sergey Miryanov in :gh:`100926`).

Reviewer 2: ```suggestion
* Move :func:`ctypes.POINTER` types cache from a global internal cache
  (``_pointer_type_cache``) to the :attr:`ctypes._CData.__pointer_type__`
  attribute of the corresponding :mod:`ctypes` types.
  This will stop the cache from growing without limits in some situations.
  (Contributed by Sergey Miryanov in :gh:`100926`).
```

  :func:`codecs.open` is now deprecated. Use :func:`open` instead.
  (Contributed by Inada Naoki in :gh:`133036`.)

* :mod:`ctypes`:
  Calling :func:`ctypes.POINTER` on a string is deprecated.
  Use :ref:`ctypes-incomplete-types` for self-referential structures.

>   Also, ``ctypes._pointer_type_cache`` is deprecated as a courtesy to
>   existing users of this internal API.

Reviewer 2: ```suggestion
  Also, the internal ``ctypes._pointer_type_cache`` is deprecated.
```
